File: Minimum number of jumps - GFG/minimum-number-of-jumps.java
//{ Driver Code Starts
import java.lang.*;
import java.io.*;
import java.util.*;
class GFG
 {
	public static void main (String[] args) throws IOException
	 {
	 
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in)); 
        int t = Integer.parseInt(br.readLine()); 

        while(t-- > 0){
            int size = Integer.parseInt(br.readLine());
            String[] arrStr = ((String)br.readLine()).split("\\s+");
            int[] arr= new int[size];
            for(int i = 0;i<size;i++){
                arr[i] = Integer.parseInt(arrStr[i]);
            }
            System.out.println(new Solution().minJumps(arr));
        }
	 }
	 
}

// } Driver Code Ends

class Solution{
    static int minJumps(int[] arr){
        // your code here
        int count=0;
        int start=0;
        int end=0;
        for(int i=0;i<arr.length-1;i++){
            start=Math.max(start,i+arr[i]);
            if(end==i){
                end=start;
                count+=1;
            }
            if(end==i && arr[i]==0){
            return -1;
        }
    }
    return count;
}
}

File: Minimum number of jumps - GFG/minimum-number-of-jumps.py
#User function Template for python3
class Solution:
    def minJumps(self, arr, n):
        count=end=start=0
        for i in range(len(arr)-1):
            start=max(start,i+arr[i])
            if end==i:
                end=start
                count+=1
            if arr[i]==0 and end==i:
                return -1
        return count


#{ 
 # Driver Code Starts
#Initial Template for Python 3
if __name__ == '__main__':
    T=int(input())
    for i in range(T):
        n = int(input())
        Arr = [int(x) for x in input().split()]
        ob = Solution()
        ans = ob.minJumps(Arr,n)
        print(ans)
# } Driver Code Ends